### PR TITLE
codeworld-api: Disable warnings on a per-module basis, fix some

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

File: codeworld-api/src/CodeWorld/Driver.hs [hunk 1]
```diff
@@ -1089,7 +1095,7 @@ getMousePos canvas = do
 onEvents :: Element -> (Event -> IO ()) -> IO ()
 onEvents canvas handler = do
     Just window <- currentWindow
-    on window keyDown $ do
+    _ <- on window keyDown $ do
```
Comment (sjakobi) [inline codeworld-api/src/CodeWorld/Driver.hs]: I thought it was interesting that the discarded result has type `IO ()`. Smells funny to me. But maybe it's fine.

```
src/CodeWorld/Driver.hs:1098:5: error: [-Wunused-do-bind, -Werror=unused-do-bind]
    A do-notation statement discarded a result of type ‘IO ()’
    Suppress this warning by saying
      ‘_ <- on window keyDown
              $ do code <- getKeyCode =<< event
                   let keyName = keyCodeToText code
                   when (keyName /= "")
                     $ do liftIO $ handler (KeyPress keyName)
                          preventDefault
                          ....
                   key <- getKey =<< event
                   ....’
     |
1098 |     on window keyDown $ do
     |     ^^^^^^^^^^^^^^^^^^^^^^...
```

File: codeworld-api/src/CodeWorld/Driver.hs [hunk 2]
```diff
@@ -1835,8 +1840,6 @@ unsafeCollaborationOf numPlayers initial step event draw = do
     dhash <- getDeployHash
     let token = PartialToken dhash
     runGame token numPlayers initial step event draw
-  where
-    token = NoToken
```
Comment (sjakobi) [inline codeworld-api/src/CodeWorld/Driver.hs]: I hope I have removed the right one of the two definitions.

File: codeworld-api/src/CodeWorld/Driver.hs (hunk 1 above)
Comment (cdsmith) [reply codeworld-api/src/CodeWorld/Driver.hs]: This is expected.  `on` registers the listener, and returns an action to unregister it.  We don't care about the unregister action.

File: codeworld-api/src/CodeWorld/Driver.hs (hunk 2 above)
Comment (cdsmith) [reply codeworld-api/src/CodeWorld/Driver.hs]: I'm not sure anyone will notice the difference.  If the token is `PartialToken dhash`, then the server will prevent anyone from joining a multiplayer game session if they are running a different binary.  That's usually desirable, so let's go with this.  Is NoToken used anywhere, then?

Comment (sjakobi) [reply codeworld-api/src/CodeWorld/Driver.hs]: > Is NoToken used anywhere, then?

I've removed it now.

File: codeworld-api/src/CodeWorld/Color.hs [hunk 3]
```diff
@@ -82,15 +79,15 @@ toHSL _ = Nothing
 mixed :: [Color] -> Color
 mixed colors = go 0 0 0 0 0 colors
   where go rr gg bb aa n ((fenceColor -> RGBA r g b a) : cs) =
-            go (rr + r^2 * a) (gg + g^2 * a) (bb + b^2 * a) (aa + a) (n + 1) cs
+            go (rr + r*r * a) (gg + g*g * a) (bb + b*b * a) (aa + a) (n + 1) cs
```
Comment (cdsmith) [inline codeworld-api/src/CodeWorld/Color.hs]: Nit-picking: It's a little confusing to write `r*r * a`.  Just `r * r * a` makes the structure clearer.

Comment (sjakobi) [reply codeworld-api/src/CodeWorld/Color.hs]: Fixed!

**Pull request merged**